rts/motoko-rts/src/types.rs

```diff
@@ -226,11 +226,13 @@ impl Array {
     }
 
     pub unsafe fn get(self: *mut Self, idx: u32) -> SkewedPtr {
```

Comment:
@osa1 I wonder if `*const Self` would be appropriate here...

Reply:
@ggreif these methods currently all have `*mut Self` receiver because until recently it wasn't possible to call a `*const Self` method on a `*mut` value, see https://github.com/rust-lang/rust/issues/80258. Without the fix for that issue it's quite noisy to use these methods as we need casts `x as *const _` everywhere.

I fixed that issue in https://github.com/rust-lang/rust/issues/80258 so we just need to update rustc.

Reply:
Not many people seem to use raw pointers in Rust...

Reply:
MMTk is also implemented in Rust, I'd expect it to use raw pointers in the GC code. Would be interesting to look at how they use raw pointers.

Reply:
Maybe we can prune this list then?
``` rust
// TODO (osa): Some of these are stabilized, we need to update rustc
#![feature(
    arbitrary_self_types,
    panic_info_message,
    assoc_char_funcs,
    core_intrinsics,
    ptr_offset_from
)]
```

Reply:
Yeah most of those should also be gone with an update.

Reply:
Yay, [here it comes](https://blog.rust-lang.org/2021/07/29/Rust-1.54.0.html)!